packages/next/src/server/lib/router-utils/typegen.ts

```diff
@@ -739,7 +739,7 @@ export function generateValidatorFileStrict(
 
   if (appPageValidations) {
     typeDefinitions += `type AppPageConfig<Route extends AppRoutes = AppRoutes> = {
-  default: React.ComponentType<PageProps<Route>> | ((props: PageProps<Route>) => React.ReactNode | Promise<React.ReactNode> | never | void | Promise<void>)
+  default: React.JSXElementConstructor<PageProps<Route>>
```

Comment:
`JSXElementConstructor` is what `ComponentType` should've been. It's the type you can use in the JSX factory. No other gimmicks like displayName, propTypes etc which make assignability harder to express.